internal/godplugin/job/wrapped_chart.go

```diff
@@ -0,0 +1,211 @@
+package job
+
+import (
+	"fmt"
+	"github.com/l2isbad/go.d.plugin/internal/pkg/charts"
+)
+
+func newWrappedChart(chart charts.Chart, hook baseConfHook, prio int) *wrappedChart {
+	p := &wrappedChart{
+		item:     chart,
+		hook:     hook,
+		priority: prio,
+		flags:    chartFlags{push: true},
+	}
+	for _, v := range chart.Dims {
+		p.dimensions = append(p.dimensions, newWrappedDim(v))
+	}
+	for _, v := range chart.Vars {
+		p.variables = append(p.variables, &wrappedVar{v})
+	}
+	return p
+}
+
+type wrappedChart struct {
+	item          charts.Chart
```

Comment:
`chart *charts.Chart`
I think ptr is better because plugin may change the chart.

Reply:
> plugin may change the chart.

```go
func (w *wrappedChart) SetID(s string) {
    w.item.ID = s
}
```